src/ontology/mondo-ingest.Makefile

```diff
@@ -515,24 +515,46 @@ sync-subclassof-%:
 	$(MAKE) $(REPORTDIR)/$*.subclass.direct-in-mondo-only.tsv
 
 # Side effects: Deletes SOURCE.subclass.direct-in-mondo-only.tsv's from which the combination is made.
-$(REPORTDIR)/sync-subClassOf.direct-in-mondo-only.tsv: $(foreach n,$(ALL_COMPONENT_IDS), sync-subclassof-$(n))
+#$(REPORTDIR)/sync-subClassOf.direct-in-mondo-only.tsv: $(foreach n,$(ALL_COMPONENT_IDS), sync-subclassof-$(n))
+$(REPORTDIR)/sync-subClassOf.direct-in-mondo-only.tsv: $(foreach n,$(ALL_COMPONENT_IDS), $(REPORTDIR)/$(n).subclass.direct-in-mondo-only.tsv)
 	python3 $(SCRIPTSDIR)/sync_subclassof_collate_direct_in_mondo_only.py --outpath $@
 
 $(REPORTDIR)/sync-subClassOf.confirmed.tsv: $(foreach n,$(ALL_COMPONENT_IDS), $(REPORTDIR)/$(n).subclass.confirmed.robot.tsv)
 	awk '(NR == 1) || (NR == 2) || (FNR > 2)' $(REPORTDIR)/*.subclass.confirmed.robot.tsv > $@
 
-# todo: Until ODK has oaklib >= 0.5.20, need '$(MAKE) pip-oaklib' below https://github.com/INCATools/ontology-development-kit/issues/936
-$(REPORTDIR)/%.subclass.added.robot.tsv $(REPORTDIR)/%.subclass.confirmed.robot.tsv $(REPORTDIR)/%.subclass.direct-in-mondo-only.tsv $(REPORTDIR)/%.subclass.added-obsolete.robot.tsv: tmp/mondo-ingest.db tmp/mondo.db tmp/mondo.sssom.tsv
+#$(REPORTDIR)/%.subclass.direct-in-mondo-only.tsv: tmp/mondo-ingest.db tmp/mondo.db tmp/mondo.sssom.tsv
+$(REPORTDIR)/%.subclass.direct-in-mondo-only.tsv: tmp/mondo-ingest.db tmp/mondo.db tmp/mondo.sssom.tsv
+	$(call sync_subclassof, $*)
+#$(REPORTDIR)/%.subclass.added.robot.tsv:
+#$(REPORTDIR)/%.subclass.confirmed.robot.tsv:
+#$(REPORTDIR)/%.subclass.direct-in-mondo-only.tsv:
+#$(REPORTDIR)/%.subclass.added-obsolete.robot.tsv:
+
+define sync_subclassof
```

Comment:
## Stuck at the moment
First, read how I got here: https://github.com/monarch-initiative/mondo-ingest/pull/397#discussion_r1467102700. Summary: I have a fresh repo now.

With fresh repo and using local `make` instead of ODK, now I'm getting this error message.

I conferred with my lord ChatGPT ([chat](https://chat.openai.com/share/3cc419dc-eaff-4d8d-9220-706721b7aaf0)) and it thinks my syntax for a multi target rule is correct. I mean, it was working before.

IMO I think the `make` error message is misleading.

I wonder if it might help to send an email to help-make@gnu.org.

I tried to resolve the error by making targets for each one of the targets in my multi-target rule, e.g. `$(REPORTDIR)/%.subclass.direct-in-mondo-only.tsv`. I have it call the code for what used to by my multi-target rule, which is now a function called `sync_subclassof`. That didn't work. However, if I make this goal:

`reports/doid.subclass.direct-in-mondo-only.tsv: tmp/mondo-ingest.db tmp/mondo.db tmp/mondo.sssom.tsv`

Now I no longer get the error: `Bug: no goal for reports/doid.subclass.direct-in-mondo-only.tsv`

Now the error I get is: `make: *** No rule to make target `tmp/mondo-ingest.db', needed by `reports/doid.subclass.direct-in-mondo-only.tsv'.  Stop.`

If I comment out those dependencies `tmp/mondo-ingest.db tmp/mondo.db tmp/mondo.sssom.tsv`, then the goal will run... But obviously we need those.

Anywhere that's where I had to leave off for today.

Reply:
Dont worry about this, I will deal with this problem once I have my computer running. I would suggest you focus on ICD 11